mrmustard/training/parameter.py

```diff
@@ -174,17 +178,27 @@ def __init__(self, value: Any, name: str, owner: Optional[str] = None) -> None:
         self._owner = owner
 
 
+class Unitary(Trainable):
+    """Unitary trainable. Uses :meth:`training.parameter_update.update_unitary`."""
+
+    def __init__(self, value: Any, name: str, owner: Optional[str] = None) -> None:
+        self._value = value_to_trainable(value, None, name)
+        self._name = name
+        self._owner = owner
+
+
 class Constant(Parameter):
     """Constant parameter. It belongs to the autograd backend but remains fixed
     during any optimization procedure
     """
 
     def __init__(self, value: Any, name: str, owner: Optional[str] = None) -> None:
-        self._value = (
-            value
-            if math.from_backend(value) and not math.is_trainable(value)
-            else math.new_constant(value, name)
-        )
+        if math.from_backend(value) and math.is_trainable(value):
+            self._value = value
+        elif type(value) in [list, int, float]:
+            self._value = math.new_variable(value, bounds=None, name=name)
```

Comment:
this is the `Constant` class, why are we making a variable?

Reply:
You are right. I forgot why I changed here. I just use the previous version of this part.